src/Adapter/Product/Update/ProductSupplierUpdater.php

```diff
@@ -131,7 +140,10 @@ public function associateSuppliers(ProductId $productId, array $supplierIds): ar
         // We should always create an association not related to a combination
         $combinationIds = [new NoCombinationId()];
         if ($productType->getValue() === ProductType::TYPE_COMBINATIONS) {
-            $combinationIds = array_merge($combinationIds, $this->combinationRepository->getCombinationIds($productId));
+            $combinationIds = array_merge(
```

Comment:
As we discussed earlier this one worries me a bit, I'm gonna work on suppliers multishop so I think it's preferable to wait to finish this PR until it's done This way we'll have tests to ensure the behaviour in multishop and we won't risk breaking some things with this refacto